### PR TITLE
gevent 23.9.1

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-3024](https://anaconda.atlassian.net/browse/PKG-3024) 
- [Upstream repository](https://github.com/gevent/gevent/tree/23.9.1)
- Changelog: https://github.com/gevent/gevent/blob/23.9.1/CHANGES.rst
- Requirements:
  - https://github.com/gevent/gevent/blob/23.9.1/pyproject.toml
  - https://github.com/gevent/gevent/blob/23.9.1/setup.py

### Explanation of changes:

- Update `script`
- Remove `m2-patch  # [win]` as patches are only for `unix`
- Remove `0003-Do-not-use-hard-coded-path-for-file-command.patch`
- Update pinnings in `host`: `cython >=3.0`, `greenlet 3.0.*`
- Update `greenlet` pinnings in `run`

### Notes:

- to adress CVE-2023-41419

[PKG-3024]: https://anaconda.atlassian.net/browse/PKG-3024?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,27 +19,22 @@ build:
...
   host:
-    # cffi 1.4.0 is the general minimum version
-    # With Python 3 version 1.11.3 is minimum
-    # On windows 1.11.5 is the minimum
-    # Rather than special casing each of these use a single minimum version
     - cffi >=1.12.3
     - pip
     - python
-    - cython
+    - cython >=3.0
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
    - cython >=3.0.2
```

**Review by lorepirri — COMMENTED** [review]
something is not available for `linux-s390x`

```
                   INFO   conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-s390x:         
                          {"python[version='>=3.10,<3.11.0a0|>=3.11,<3.12.0a0|>=3.12,<3.13.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a…
                          'python', "python[version='>=3.12,<3.13.0a0']", "_libgcc_mutex[version='*|0.1',build=main]",                      
                          "libgcc-ng[version='>=7.5.0']", "expat[version='>=2.5.0,<3.0a0']", "libgcc-ng[version='>=11.2.0|>=7.5.0']",       
                          "libgcc-ng[version='>=11.2.0']",                                                                                  
                          "python[version='>=3.10,<3.11.0a0|>=3.11,<3.12.0a0|>=3.12,<3.13.0a0|>=3.8,<3.9.0a0|>=3.7,<3.8.0a0|>=3.9,<3.10.0a0…
                          "python[version='>=3.6']",                                                                                        
                          "python[version='>=3.10,<3.11.0a0|>=3.11,<3.12.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0']", 'setuptools',
                          "python[version='>=3.10,<3.11.0a0|>=3.11,<3.12.0a0|>=3.12,<3.13.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0|>=3.7,<3.8.0a0…
                          "python[version='>=3.10,<3.11.0a0|>=3.12,<3.13.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0|>=3.11,<3.12.0a0']"}            
12/18/23 10:05:48  ERROR  Command failed with exit code 1                                                                                   

```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  sha256: 72c002235390d46f94938a96920d8856d4ffd9ddf62a303a0d7c118894097e34
   patches:   # [unix]
     - 0001-Allow-include-flags-to-be-present-in-CFLAGS.patch        # [unix]
     - 0002-Allow-library-flags-to-be-present-in-LDFLAGS.patch       # [unix]
-    - 0003-Do-not-use-hard-coded-path-for-file-command.patch        # [unix]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:16]: Could you re-sync them with conda-forge please?

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Fixed

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:16]: Thanks for pointing me out. I refreshed patches

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,48 +1,40 @@
...
-    # On windows 1.11.5 is the minimum
-    # Rather than special casing each of these use a single minimum version
     - cffi >=1.12.3
     - pip
     - python
-    - cython
+    - cython >=3.0.2
     - setuptools
-    - greenlet 2.0.*
+    - greenlet 3.0.*
```
Comment (boldorider4) [inline recipe/meta.yaml]: shouldn't it be an exact pinning?

**Review by boldorider4 — APPROVED** [review]
LGTM except minor comment about exact pinning.

File: recipe/meta.yaml (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Yep, I think `greenlet 3.0` will be fine

**Comment by skupr-anaconda** [comment]
> something is not available for `linux-s390x`
> 
> ```
>                    INFO   conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-s390x:         
>                           {"python[version='>=3.10,<3.11.0a0|>=3.11,<3.12.0a0|>=3.12,<3.13.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a…
>                           'python', "python[version='>=3.12,<3.13.0a0']", "_libgcc_mutex[version='*|0.1',build=main]",                      
>                           "libgcc-ng[version='>=7.5.0']", "expat[version='>=2.5.0,<3.0a0']", "libgcc-ng[version='>=11.2.0|>=7.5.0']",       
>                           "libgcc-ng[version='>=11.2.0']",                                                                                  
>                           "python[version='>=3.10,<3.11.0a0|>=3.11,<3.12.0a0|>=3.12,<3.13.0a0|>=3.8,<3.9.0a0|>=3.7,<3.8.0a0|>=3.9,<3.10.0a0…
>                           "python[version='>=3.6']",                                                                                        
>                           "python[version='>=3.10,<3.11.0a0|>=3.11,<3.12.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0']", 'setuptools',
>                           "python[version='>=3.10,<3.11.0a0|>=3.11,<3.12.0a0|>=3.12,<3.13.0a0|>=3.9,<3.10.0a0|>=3.8,<3.9.0a0|>=3.7,<3.8.0a0…
>                           "python[version='>=3.10,<3.11.0a0|>=3.12,<3.13.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0|>=3.11,<3.12.0a0']"}            
> 12/18/23 10:05:48  ERROR  Command failed with exit code 1                                                                                   
> ```

Fixed

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by lorepirri — APPROVED** [review]

**Pull request merged**